### PR TITLE
WIP: Create a Docker-Compose for the project

### PR DESCRIPTION
**Pull request by DiSiqueira**
- Creates a multi-stage Dockerfile
- Creates a docker-compose.yml file for the project
- Update makefile to use docker-compose 
- Update circle.yml
- Update the example in the README file

All the hard job should be done, but I'll not be able to test it today, so I'll ask you to wait for a little before merging it.

Please give me the feedback if you guys liked the approach or if you want me to change something, and If there is anything else just let me know, I'll be happy to help.

### CODE REVIEW

**Comment by lferreira** [comment]
Hi @DiSiqueira, We're really happy with your contributions 🚀

File: Makefile [hunk 1]
```diff
@@ -13,18 +10,10 @@ install:
...
 docker_build:
-	docker build -t ${DOCKER_REPO}/x9:build -f Dockerfile.build .
-
-docker_image: docker_build
-	docker run --rm --entrypoint /bin/sh -v ${PWD}:/out:rw ${DOCKER_REPO}/x9:build -c "cp /go/bin/x9 /out/x9"
-	docker build -t ${DOCKER_REPO}/x9 .
-
-docker_tag:
-	docker tag ${DOCKER_REPO}/x9 ${DOCKER_REPO}/x9:${DOCKER_IMAGE_VERSION}
+	docker-compose build
```
Comment (lorieri) [inline Makefile]: I have intention to use docker-compose, could you separate it from regular builds ?
And it looks strange here under "docker-tag", is it correct ?

File: Makefile [hunk 2]
```diff
@@ -13,18 +10,10 @@ install:
...
 
 docker_run:
-	docker run -d --rm ${DOCKER_REPO}/x9:${DOCKER_IMAGE_VERSION}
+	docker-compose up -d
```
Comment (lorieri) [inline Makefile]: same here

File: Makefile [hunk 3]
```diff
@@ -13,18 +10,10 @@ install:
...
 
 docker_push:
-	docker push ${DOCKER_REPO}/x9:${DOCKER_IMAGE_VERSION}
-
+	docker-compose push
```
Comment (lorieri) [inline Makefile]: same here

File: README.md [hunk 4]
```diff
@@ -76,15 +76,11 @@ $ make build
...
--e REDIS_SERVER=redis:6379 \
--e AWS_ACCESS_KEY_ID=XXX \
--e AWS_SECRET_ACCESS_KEY="XXX" \
--e SLACK_WEBHOOK_URL="https://myslack...." \
-myrepo/x9
+```bash
+$ EXPORT AWS_ACCESS_KEY_ID=XXX
+$ EXPORT AWS_SECRET_ACCESS_KEY="XXX"
+$ EXPORT SLACK_WEBHOOK_URL="https://myslack...."
+$ make docker_run
```
Comment (lorieri) [inline README.md]: I believe you will need a "make docker_compose_run" or something

File: README.md [hunk 5]
```diff
@@ -100,7 +96,8 @@ $ curl localhost:6969/api
 |ALERT_TIMEFRAME|1200|Interval between checks|
 |SERVICE_PORT|6969|Webserver port|
 |REGIONS|"sa-east-1,us-east-1"|Regions were to run "describe-instances"|
-
+|DOCKER_IMAGE_VERSION|latest|Version that will be used to tag the final image|
+|DOCKER_REPO|myrepo|Repository that will be used to tag the final image|
```
Comment (lorieri) [inline README.md]: those 2 opt are not necessary for the app to run

my suggestion is separate it in another section like 

# Build options

(same as https://github.com/VivaReal/x9/pull/5#pullrequestreview-72548729 )

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    - '6969:6969'
+    depends_on:
+      - redis
+
+  # Redis Database
+  redis:
+    image: redis
+    ports:
+    - '6379'
+
```
Comment (lorieri) [inline docker-compose.yml]: thanks :)
Even though I not intent to use docker-compose, some people will be happy with this :)

**Pull request closed without merging**